### ISSUE TITLE
update map syntax works for more key types than just atoms

### ISSUE THREAD

**Issue by nhooyr** [doomspork/elixir-school#1091]
http://elixirschool.com/lessons/basics/collections/#maps

At the bottom it states

> Another interesting property of maps is that they provide their own syntax for updating and accessing atom keys:
```elixir
iex> map = %{foo: "bar", hello: "world"}
%{foo: "bar", hello: "world"}
iex> %{map | foo: "baz"}
%{foo: "baz", hello: "world"}
iex> map.hello
"world"
```

However, the updating syntax works for more than just atom keys, e.g.

```elixir
iex> %{map | "STRINGKEY" => :new_val}
```

**Comment by doomspork** [comment]
Howdy @nhooyr for the most part we don't attempt to cover _every_ possibility, rather we aim to provide a foundation from which to build on.  I'm not sure we need to make changes to call out each supported key type but pull requests are welcome if you think a change could be made 👍

**Comment by nhooyr** [comment]
Its misleading because it implies that the syntax only works for updating atom keys.

**Comment by doomspork** [comment]
@nhooyr perhaps we should then expand on the commentary around that block?  My concern is that more than just atoms and strings are valid map keys, where do we draw the line with examples?

```elixir
# You can use numbers
iex> map = %{1 => "one"}
%{1 => "one"}
iex> %{map | 1 => "two"}
%{1 => "two"}

# You can use modules (which are admittedly atoms under the hood)
iex> map = %{Test: "foo"}
%{Test: "foo"}
iex> %{map | Test: "bar"}
%{Test: "bar"}
```

**Comment by nhooyr** [comment]
It's not a issue with the examples, its an issue with the text. Examples don't have to consider every case but the text should because otherwise it is misleading. A fix for this is included in #1094
